yaksh/views.py

```diff
@@ -517,11 +517,16 @@ def check(request, q_id, attempt_num=None, questionpaper_id=None):
                         if question.type == 'code' else None
         correct, result = paper.validate_answer(user_answer, question, json_data)
         if correct:
+            new_answer.marks = (question.points * result['weightage'] / 
```

Comment:
I had already fixed this in a minor commit, It did not get pushed. I apologise. This has been fixed